dash/test/algorithm/SortTest.cc

```diff
@@ -0,0 +1,331 @@
+
+  rand_range(array.begin(), array.end());
+
+  array.barrier();
+
+  perform_test(array.begin(), array.end());
+}
+
+TEST_F(SortTest, ArrayOfDoubles)
+{
+  typedef double                 Element_t;
+  typedef dash::Array<Element_t> Array_t;
+
+  LOG_MESSAGE("SortTest.ArrayOfDoubles: allocate array");
+  // Initialize global array:
+  Array_t array(num_local_elem * dash::size());
+
+  rand_range(array.begin(), array.end());
+
+  // Wait for all units
+  array.barrier();
+
+  perform_test(array.begin(), array.end());
+}
+
+TEST_F(SortTest, MatrixBlockedRow)
+{
+  using value_t     = int32_t;
+  using block_pat_t = dash::BlockPattern<2, dash::ROW_MAJOR>;
+  using narray_t =
+      dash::NArray<value_t, 2, dash::default_index_t, block_pat_t>;
+
+  size_t team_size = dash::Team::All().size();
+  size_t extent_x  = 17;
+  size_t extent_y  = 5 + team_size * 3;
+  size_t size      = extent_x * extent_y;
+
+  // Ceil division
+  size_t block_size_x = extent_x;
+  size_t block_size_y = dash::math::div_ceil(extent_y, team_size);
+
+  LOG_MESSAGE(
+      "ex: %d, ey: %d, bsx: %d, bsy: %d",
+      static_cast<int>(extent_y),
+      static_cast<int>(extent_x),
+      static_cast<int>(block_size_y),
+      static_cast<int>(block_size_x));
+
+  block_pat_t pat_blocked_row{
+      dash::SizeSpec<2>(extent_y, extent_x),
+      dash::DistributionSpec<2>(dash::BLOCKED, dash::NONE),
+      dash::TeamSpec<2>(dash::Team::All()),
+      dash::Team::All()};
+
+  narray_t mat{pat_blocked_row};
+
+  rand_range(mat.begin(), mat.end());
+
+  mat.barrier();
+
+  perform_test(mat.begin(), mat.end());
+}
+
+TEST_F(SortTest, ArrayOfPoints)
+{
+  typedef Point                  Element_t;
+  typedef dash::Array<Element_t> Array_t;
+
+  LOG_MESSAGE("SortTest.ArrayOfPoints: allocate array");
+  // Initialize global array:
+  Array_t array(num_local_elem * dash::size());
+
+  static std::uniform_int_distribution<int32_t> distribution(-1000, 1000);
+  static std::random_device                     rd;
+  static std::mt19937 generator(rd() + array.team().myid());
+
+  dash::generate(array.begin(), array.end(), []() {
+    return Point{distribution(generator), distribution(generator)};
+  });
+
+  array.barrier();
+
+  dash::sort(array.begin(), array.end(), [](const Point& p) { return p.x; });
+
+  if (dash::myid() == 0) {
+    for (auto it = array.begin() + 1; it < array.end(); ++it) {
+      auto const a = static_cast<const Element_t>(*(it - 1));
+      auto const b = static_cast<const Element_t>(*it);
+
+      EXPECT_FALSE_U(b < a);
+    }
+  }
+}
+
+template <
+    class ValueType,
+    typename std::enable_if<std::is_floating_point<ValueType>::value>::type* =
+        nullptr>
+static void expect_equals(ValueType const e, ValueType const a)
+{
+  if (std::is_same<decltype(e), double>::value) {
+    EXPECT_DOUBLE_EQ(e, a) << "Unit " << dash::myid().id;
+  }
+  else if (std::is_same<decltype(e), float>::value) {
+    EXPECT_FLOAT_EQ(e, a) << "Unit " << dash::myid().id;
+  }
+}
```

Comment:
Interesting, `EXPECT_EQ_U` should dispatch to the correct equality testing macro, see https://github.com/dash-project/dash/blob/development/dash/test/TestBase.h#L123

Reply:
GoogleTest provides specific macros to compare floats (nearly equal). Our (custom) solution which is different from Google's provided solution here did not work for me here at least. That is why I explicitly call the "correct" GoogleTest macro. I will fix that and provide a separate pull request for all unit tests.

Reply:
Furthermore we should not rely on internal GoogleTest APIs which may change.